src/mobile/context/auth/provider.tsx

```diff
@@ -0,0 +1,92 @@
+import { useRouter, useSegments } from 'expo-router'
+import {
+  createContext,
+  PropsWithChildren,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+
+type SelfUser =
+  | {
+      authenticated: boolean
+      email: string
```

Comment:
can you make these nullable or something? maybe I need to get used to this.. typescript feels funky :(

Reply:
I'm still not sure what the best way to do this is. I think I'll have a better idea as I create the auth system and have to use user data in the screens.

Reply:
The way this is laid out is inspired by django, since you can always count on checking `request.user.is_authenticated`.

Reply:
yeaaaaaaaaaah, i have no idea if this is a good/bad way, leave as is, is fine! ty